plugin/pkg/scheduler/schedulercache/cache.go

```diff
@@ -343,6 +363,18 @@ func (cache *schedulerCache) cleanupAssumedPods(now time.Time) {
 		if !ok {
 			panic("Key found in assumed set but not in podStates. Potentially a logical error.")
 		}
+		var bindingDone bool
+		select {
+		case <-ps.waitsBinding:
+			bindingDone = true
+		default:
+			bindingDone = false
+		}
+		if !bindingDone {
+			glog.Warningf("Cant expire cache for pod %v/%v. Binding is still in progress.",
+				ps.pod.Namespace, ps.pod.Name)
+			continue
+		}
```

Comment:
I completely don't understand this logic. Finished binding is not the signal for expiring pod - if the binding succeeded, then we want this to be assumed until we observe it via watch, otherwise, it will be forget anyway.

So this PR is incorrect.